src/renderer/views/Playlist/Playlist.js

```diff
@@ -416,10 +429,37 @@ export default defineComponent({
 
       this.isLoading = false
     },
+
     showUserPlaylistNotFound() {
       showToast(this.$t('User Playlists.SinglePlaylistView.Toast.This playlist does not exist'))
     },
 
+    getPlaylistItemsWithDuration() {
+      const modifiedPlaylistItems = deepCopy(this.playlistItems)
+      let anyVideoMissingDuration = false
+      modifiedPlaylistItems.forEach(video => {
+        if (videoDurationPresent(video)) { return }
+
+        const videoHistory = this.$store.getters.getHistoryCacheById[video.videoId]
+        if (typeof videoHistory !== 'undefined') {
+          video.lengthSeconds = videoHistory.lengthSeconds
```

Comment:
You need to check if the duration is valid here too, as just like with the playlist, just because a video exists doesn't mean all the information is there. Example: Manually marking a video as watched through the 3 dots menu on the shorts channel tab or in a playlist that only contains shorts.

Reply:
done!